core/src/main/java/com/netease/arctic/iceberg/optimize/DeleteFilter.java

```diff
@@ -285,4 +301,54 @@ private static Schema fileProjection(Schema tableSchema, Schema requestedSchema,
 
     return new Schema(columns);
   }
+
+
+  private class CloseableIterableWithOtherCloseable<T> implements CloseableIterable<T> {
```

Comment:
This is the same implementation as inner class of CombinedDeleteFilter, could it be reused?

Reply:
There are some differences between the two implementations，but reuse is not impossible。I will make some changes